From df2000bc01fdded01353f7535068f9153c70abcb Mon Sep 17 00:00:00 2001
From: Masaru Nagai <masaru.nagai.vx@renesas.com>
Date: Wed, 31 May 2017 13:20:34 +0900
Subject: [PATCH 189/909] ravb: Fix interrupt cause bit overwrite in
 ravb_ptp_interrupt

commit 029f9e9bcfb1666dd2f63d85292d1000d63f4d7d from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Signed-off-by: Masaru Nagai <masaru.nagai.vx@renesas.com>
Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/renesas/ravb_ptp.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/renesas/ravb_ptp.c b/drivers/net/ethernet/renesas/ravb_ptp.c
index 44539314963f..95678c2cbcf0 100644
--- a/drivers/net/ethernet/renesas/ravb_ptp.c
+++ b/drivers/net/ethernet/renesas/ravb_ptp.c
@@ -312,7 +312,7 @@ irqreturn_t ravb_ptp_interrupt(struct net_device *ndev)
 		ptp_clock_event(priv->ptp.clock, &event);
 
 		result = IRQ_HANDLED;
-		gis &= ~GIS_PTCF;
+		ravb_write(ndev, ~GIS_PTCF, GIS);
 	}
 	if (gis & GIS_PTMF) {
 		struct ravb_ptp_perout *perout = priv->ptp.perout;
@@ -323,12 +323,9 @@ irqreturn_t ravb_ptp_interrupt(struct net_device *ndev)
 		}
 
 		result = IRQ_HANDLED;
-		gis &= ~GIS_PTMF;
+		ravb_write(ndev, ~GIS_PTMF, GIS);
 	}
 
-	gis &= ~GIS_RESERVED;
-	ravb_write(ndev, gis, GIS);
-
 	return result;
 }
 
-- 
2.17.1

